ignite/pkg/protoanalysis/protoutil/formatter.go

```diff
@@ -0,0 +1,538 @@
+package protoutil
+
+// An alternative formatter that doesn't try to be as smart as the one
+// currently used.
+
+// NOTES:
+//  - Doesn't handle C-style comments (prints as regular) or inline ones.
+//  - Doesn't handle proto2 syntax.
+//  - Can make it detect duplicate imports/double syntax/package and fields
+//	  with weird sequencing.
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/emicklei/proto"
+)
+
+const (
+	srv     = "service"
+	enum    = "enum"
+	message = "message"
+	oneof   = "oneof"
+)
+
+// Formatter visits a Proto and writes formatted source.
+type Formatter struct {
+	w                  io.Writer
+	indentSeparator    string
+	lastStmt, currStmt string
+	indentLevel        int
+}
+
+// NewFormatter returns a new Formatter. Only the indentation separator is configurable.
+func NewFormatter(writer io.Writer, indentSeparator string) *Formatter {
+	return &Formatter{w: writer, indentSeparator: indentSeparator}
+}
+
+// Format visits all proto elements and writes formatted source.
+func (f *Formatter) Format(p *proto.Proto) {
```

Comment:
This method seems to be used only in tests, do we really need it ?

Reply:
These are hackathon relics, the current output of protobuf files after modifications has certain opinions. This was developed (frantically!) in the last days as an alternative in order for us to control the formatting if required. It isn't used currently but might in the future? I'll remove it if not required.

Reply:
I notice that you already use `protofmt` to format proto files, like [here](https://github.com/ignite/cli/pull/3109/files#diff-f0d5b5827b6fabef25dda32d246f499fc4fb3b970e976aa4c8e73f91aa7be6edR33).

If the documentation is up-do-date :
```go
// An alternative formatter that doesn't try to be as smart as the one
// currently used.

// NOTES:
//  - Doesn't handle C-style comments (prints as regular) or inline ones.
//  - Doesn't handle proto2 syntax.
//  - Can make it detect duplicate imports/double syntax/package and fields
//	  with weird sequencing.
```

Can you tell me why being less smart than `protofmt` actually matters ?

Reply:
Simplicity in maintenance is probably what I was after (though I'm guessing, can't remember details). I'll just remove it for now, if custom output format is needed in the future, it'll be relatively easy to add.